vortexasdk/client.py

```diff
@@ -33,56 +24,68 @@ def get_reference(self, resource: str, id: ID) -> str:
         """Lookup reference data."""
         url = self._create_url(f'{resource}/{id}')
         response = requests.get(url)
-        return self._handle_response(response)['data']
+        return _handle_response(response)['data']
 
     def search(self, resource: str, **data) -> List:
         """Search using `resource` using `**data` as filter params."""
         url = self._create_url(resource)
 
         payload = {k: v for k, v in data.items() if v is not None}
 
-        total = self._sent_post_request(url, payload, size=1, offset=0)['total']
+        total = _send_post_request(url, payload, size=1, offset=0)['total']
 
         size = data.get('size', 1000)
         offsets = [i for i in range(0, total, size)]
 
-        pmap = map
-        # pmap = Pool(N_PARALLELISM * 2).map
-
-        send_request = lambda x: self._sent_post_request(url, payload, size, x)['data']
+        n_threads = 4
+        with ThreadPool(n_threads) as pool:
+            print(f'{total} Results to retreive.'
+                  f' Sending {len(offsets)}'
+                  f' post requests in parallel using {n_threads} threads.')
 
-        responses = list(pmap(send_request, offsets))
+            responses = pool.map(functools.partial(_send_post_request_data, url=url, payload=payload, size=size),
+                                 offsets)
 
-        flattened = [x for y in responses for x in y]
+            flattened = [x for y in responses for x in y]
 
-        assert len(flattened) == total
-        return flattened
+            assert len(flattened) == total, f'len(flattened) = {len(flattened)}, total = {total}'
```

Comment:
Is a failure of this assertion meant to give the end-user any insight into the error, or is it just for developer debugging?  If it's the latter it's fine, but if it's intended for the end-user, I don't think message doesn't help much as it references variables created in the scope of the function.  In which case something like 'number of responses is does not match number of requests` may be more insightful.